components/codec/src/byte.rs

```diff
@@ -352,6 +353,35 @@ impl MemComparableCodecHelper for DescendingMemComparableCodecHelper {
     }
 }
 
+pub trait BufferByteEncoder: BufferNumberEncoder {
+    fn write_bytes(&mut self, bs: &[u8], desc: bool) -> Result<()> {
```

Comment:
Prefer to separate them into different functions based on different `desc` value. This can eliminate one branch.